src/pybind/mgr/telemetry/tests/telemetry_upgrade_tests.md

```diff
@@ -0,0 +1,208 @@
+  but these only tests the telemetry workunits and do not run any other
+  workloads. 
+
+A simple upgrade test contains the test in the order:
+```
+├── 0-start.yaml
+├── 1-tasks.yaml
+├── upgrade-sequence.yaml
+└── workload
+```
+
+- `0-start.yaml`: This file contains the information about the ceph cluster
+  configuration (number of osds, monitors etc) for the test
+- `1-tasks.yaml`: This file contains the information of the tasks we want to run
+  on the cluster. It is here that we install an older release, then begin
+  running the given `workload` and `upgrade-sequence` in parallel.
+- `upgrade-sequence.yaml`: This file contains the steps for upgrading the
+  cluster to the latest release
+- `workloads`: A set of yaml file with workloads we want to run while upgrade is
+  in progress
+
+```
+- print: "**** done start parallel"
+- parallel:
+    - workload
+    - upgrade-sequence
+- print: "**** done end parallel"
+```
+
+The `workload` directory contains the workload yaml files just like any other
+suite and the `upgrade-sequence` is responsible for running the upgrade and
+waiting for it to complete.
+
+```
+# renamed tasks: to upgrade-sequence:
+upgrade-sequence:
+   sequential:
+   - print: "**** done start upgrade, wait"
+   ...
+       mon.a:
+         - ceph config set global log_to_journald false --force
+         - ceph orch upgrade start --image quay.ceph.io/ceph-ci/ceph:$sha1
+         - while ceph orch upgrade status | jq '.in_progress' | grep true && ! ceph orch upgrade status | jq '.message' | grep Error ; do ceph orch ps ; ceph versions ; ceph orch upgrade status ; sleep 30 ; done
+   ...
+   - print: "**** done end upgrade, wait..."
+```
+
+## Telemetry Upgrade tests
+
+The telemetry upgrade test suite verifies that the telemetry is emitting the
+correct collections after the upgrade. This integration test coverage are done
+via workunits. Workunits are basically bash scripts that run commands against a
+Ceph Cluster. 
+
+In the same manner as the upgrade test, each release branch `qa/workunits`
+directory includes telemetry test for X-2 releases. That means, we can test
+telemetry before and after the upgrade from previous two releases to the current
+one.
+
+For instance, the relevant telemetry workunits for `tentacle` release are:
+```
+qa/workunits
+├── test_telemetry_reef.sh
+├── test_telemetry_reef_x.sh
+├── test_telemetry_squid.sh
+└── test_telemetry_squid_x.sh
+```
+
+- `test_telemetry_reef.sh`, tests the presence of telemetry collection on a reef cluster before the upgrade.
+- `test_telemetry_reef_x.sh`, tests the presence of new telemetry collection on the X-version cluster after it has been upgrade from reef.
+- `test_telemetry_squid.sh`, tests the presence of telemetry collection on a squid cluster before the upgrade
+- `test_telemetry_squid_x.sh`, tests the presence of new telemetry collection on the X-version cluster after it has been upgrade from squid.
```

Comment:
Update the workunits referenced here to align with the squid example (see https://github.com/ceph/ceph/tree/squid/qa/workunits)

Reply:
thank you! done